runtime/src/bank.rs

```diff
@@ -1206,9 +1206,21 @@ impl Bank {
     }
 
     pub fn new_for_tests(genesis_config: &GenesisConfig) -> Self {
+        Self::_new_for_tests(genesis_config, false)
+    }
+
+    pub fn new_for_tests_with_secondary_indexes(genesis_config: &GenesisConfig) -> Self {
+        Self::_new_for_tests(genesis_config, true)
+    }
+
+    fn _new_for_tests(genesis_config: &GenesisConfig, enable_secondary_indexes: bool) -> Self {
```

Comment:
let's make this `fn new_for_tests_with_config(...`, wrap `enable_secondary_indexes` with a config struct like `TestsBankConfig { ... }` and dump the secondary-index-specific constructor. encoding a specific usecase bloats the API and is typically not generally useful

Reply:
So I just wanted to add this to the GenisisConfig type originally to avoid this but it enforces the Serialize/Deseralize and other traits that are not on the AccountSecondaryIndexes struct. Id perfer to just add to that if possible so let me explore what can be done and get back to you.

Reply:
take a look lmk if what I did is acceptable.